### PR TITLE
refactor: calculate class hash in devnet

### PR DESCRIPTION
**Pull request by apoorvsadana**
In the current setup, the class hashes for devnet are hard coded. This isn't really a big issue but everytime we update the Cairo compiler we need to manually change these class hashes in the devnet. This PR calculates the class hash instead of taking the hardcoded values.

### CODE REVIEW

**Review by Trantorian1 — APPROVED** [review]
All good :+1:

File: crates/client/devnet/src/classes.rs [hunk 1]
```diff
@@ -24,11 +26,23 @@ pub enum InitiallyDeclaredClass {
 }
 
 impl InitiallyDeclaredClass {
```
Comment (jbcaron) [inline crates/client/devnet/src/classes.rs:28]: just out of curiosity, why aren’t the classes definition stored compressed directly?

Comment (apoorvsadana) [reply crates/client/devnet/src/classes.rs:28]: I didn't understand the question sorry, the compressed version is stored now right?

**Comment by antiyro** [comment]
lgtm have you tested it with different scarb versions?

**Comment by apoorvsadana** [comment]
> lgtm have you tested it with different scarb versions?

Yes!

**Review by antiyro — APPROVED** [review]

File: crates/client/devnet/src/classes.rs (hunk 1 above)
Comment (jbcaron) [reply crates/client/devnet/src/classes.rs:28]: it's just that we produce uncompressed class definitions with scarb

Comment (apoorvsadana) [reply crates/client/devnet/src/classes.rs:28]: Oh ok, I am not really sure but if Scarb has that option then this could be an optimisation, yes.

**Pull request merged**